packages/store/src/StoreCore.sol

```diff
@@ -169,7 +169,7 @@ library StoreCore {
 
     // Call onSetRecord hooks (before actually modifying the state, so observers have access to the previous state if needed)
     address[] memory hooks = Hooks.get(bytes32(tableId));
-    for (uint256 i = 0; i < hooks.length; i++) {
+    for (uint256 i; i < hooks.length; i++) {
```

Comment:
is this a gas saver? if not, I kinda prefer the more explicit version

Reply:
it saves a little bit of gas! I used to prefer `i = 0` too, but got used to not initializing zero values as an easy optimization. At this point we use the non-initialize pattern in most MUD contracts, so I changed it here for consistency.